sphinx/tutorial/cylc/furthertopics/family-triggers.rst

```diff
@@ -0,0 +1,184 @@
+tasks in the ``MINERS`` family succeeding.
+
+Open your ``suite.rc`` file and change the ``[[dependencies]]`` to look like
+this:
+
+.. code-block:: cylc
+
+   [[dependencies]]
+       graph = """visit_mine => MINERS
+                  MINERS:finish-all & MINERS:succeed-any => sell_diamonds"""
+
+Then, add the following task to the ``[runtime]`` section:
+
+.. code-block:: cylc
+
+   [[sell_diamonds]]
+      script = sleep 5
+
+These changes add a ``sell_diamonds`` task to the suite which is run once
+all the ``MINERS`` tasks have finished and if any of them have succeeded.
+
+Save your changes and run your suite. You should see the new
+``sell_diamonds`` task being run once all the miners have finished and at
+least one of them has succeeded. As before, stop your suite using the "stop"
+button in the ``cylc gui``.
+
+
+Family triggering: failure
+--------------------------
+
+Cylc also allows us to trigger off failure of tasks in a particular family.
+
+We would like to add another task to close down unproductive mineshafts once
+all the miners have reported back and had time to discuss their findings.
+
+To do this we will make use of family triggers in a similar manner to before.
+
+Open your ``suite.rc`` file and change the ``[[dependencies]]`` to look like
+this:
+
+.. code-block:: cylc
+
+   [[dependencies]]
+       graph = """visit_mine => MINERS
+                  MINERS:finish-all & MINERS:succeed-any => sell_diamonds
+                  MINERS:finish-all & MINERS:fail-any => close_shafts
+                  close_shafts => !MINERS
+                  """
+
+Alter the ``[[sell_diamonds]]`` section to look like this:
+
+.. code-block:: cylc
+
+   [[close_shafts, sell_diamonds]]
+       script = sleep 5
+
+These changes add a ``close_shafts`` task which is run once all the
+``MINERS`` tasks have finished and any of them have failed. On completion
+it applies a *suicide trigger* to the ``MINERS`` family in order to allow
+the suite to shutdown.
+
+Save your changes and run your suite. You should see the new
+``close_shafts`` run should any of the ``MINERS`` tasks be in the failed
+state once they have all finished.
+
+
+Different triggers
+------------------
+
+Other types of triggers beyond those covered in the example are also available.
+
+The following types of "all" type triggers are available:
+
+* ``FAM:start-all`` - all the tasks in FAM have started
+* ``FAM:succeed-all`` - all the tasks in FAM have succeeded
+* ``FAM:fail-all`` - all the tasks in FAM have failed
+* ``FAM:finish-all`` - all the tasks in FAM have finished
+
+The following types of "any" type triggers are available:
+
+* ``FAM:start-any`` - at least one task in FAM has started
+* ``FAM:succeed-any`` - at least one task in FAM has succeeded
+* ``FAM:fail-any`` - at least one task in FAM has failed
+* ``FAM:finish-any`` - at least one task in FAM has finished
+
+
+Summary
+-------
+
+* Family triggers allow you to create dependencies on particular families.
```

Comment:
I would replace 'families' with 'defined related sets of tasks' to avoid repeating the obvious given the concept name.